subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/FileSystemFixture.groovy

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.integtests.fixtures
+
+import groovy.transform.SelfType
+import org.apache.commons.lang.RandomStringUtils
+import org.gradle.internal.os.OperatingSystem
+
+import java.nio.file.Files
+
+import static org.hamcrest.CoreMatchers.equalTo
+import static org.hamcrest.MatcherAssert.assertThat
+
+@SelfType(AbstractIntegrationSpec)
+trait FileSystemFixture {
+
+    private static def potentialLocationsOnLinux = ["/tmp", "/xfs"]
+
+    TestFileSystem createNewFS() {
```

Comment:
❓ Can we phrase this as a JUnit `@Rule` so the lifecycle of creating and cleaning up the additional FS would be handled automatically? We can move tests that deal with separate file systems to a separate integration test class.